frame/dapp-staking-v3/src/test/mock.rs

```diff
@@ -0,0 +1,182 @@
+    type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+impl pallet_balances::Config for Test {
+    type MaxLocks = ConstU32<4>;
+    type MaxReserves = ();
+    type ReserveIdentifier = [u8; 8];
+    type Balance = Balance;
+    type RuntimeEvent = RuntimeEvent;
+    type DustRemoval = ();
+    type ExistentialDeposit = ConstU128<EXISTENTIAL_DEPOSIT>;
+    type AccountStore = System;
+    type WeightInfo = ();
+}
+
+impl pallet_dapp_staking::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type Currency = Balances;
+    type SmartContract = MockSmartContract;
+    type ManagerOrigin = frame_system::EnsureRoot<AccountId>;
+    type MaxNumberOfContracts = ConstU16<10>;
+    type MaxLockedChunks = ConstU32<5>;
+    type MaxUnlockingChunks = ConstU32<5>;
+    type MinimumLockedAmount = ConstU128<MINIMUM_LOCK_AMOUNT>;
+}
+
+#[derive(PartialEq, Eq, Copy, Clone, Encode, Decode, Debug, TypeInfo, MaxEncodedLen, Hash)]
+pub enum MockSmartContract {
+    Wasm(AccountId),
+    Other(AccountId),
+}
+
+impl Default for MockSmartContract {
+    fn default() -> Self {
+        MockSmartContract::Wasm(1)
+    }
+}
+
+pub struct ExtBuilder;
+impl ExtBuilder {
+    pub fn build() -> TestExternalities {
+        let mut storage = frame_system::GenesisConfig::default()
+            .build_storage::<Test>()
+            .unwrap();
+
+        let balances = vec![1000; 9]
+            .into_iter()
+            .enumerate()
+            .map(|(idx, amount)| (idx as u64 + 1, amount))
+            .collect();
+
+        pallet_balances::GenesisConfig::<Test> { balances: balances }
+            .assimilate_storage(&mut storage)
+            .ok();
+
+        let mut ext = TestExternalities::from(storage);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+}
+
+/// Run to the specified block number.
+/// Function assumes first block has been initialized.
+pub(crate) fn run_to_block(n: u64) {
+    while System::block_number() < n {
+        DappStaking::on_finalize(System::block_number());
+        System::set_block_number(System::block_number() + 1);
+        // This is performed outside of dapps staking but we expect it before on_initialize
+        DappStaking::on_initialize(System::block_number());
+    }
+}
+
+/// Run for the specified number of blocks.
+/// Function assumes first block has been initialized.
+pub(crate) fn _run_for_blocks(n: u64) {
+    run_to_block(System::block_number() + n);
+}
+
+/// Advance blocks until the specified era has been reached.
+///
+/// Function has no effect if era is already passed.
+pub(crate) fn advance_to_era(era: EraNumber) {
+    // TODO: Properly implement this later when additional logic has been implemented
+    ActiveProtocolState::<Test>::mutate(|state| state.era = era);
+}
+
+/// Initialize first block.
+/// This method should only be called once in a UT otherwise the first block will get initialized multiple times.
+pub(crate) fn initialize_first_block() {
```

Comment:
Could we have it executed in `build` function, so it doesn't have to repeat in each UT?